test/authorization/service/role.service.test.ts

```diff
@@ -78,170 +95,226 @@ describe('test Role Service', () => {
         { provide: 'RedisCacheService', useValue: redisCacheService },
         { provide: 'SearchService', useValue: searchService },
         {
-          provide: Connection,
-          useValue: connectionMock,
+          provide: DataSource,
+          useValue: mockDataSource,
         },
       ],
     }).compile();
-    roleService = moduleRef.get<RoleService>(RoleService);
-  });
 
-  it('should get all roles', async () => {
-    rolesRepository.createQueryBuilder().returns(rolesQueryBuilder);
-    rolesQueryBuilder.getManyAndCount().returns(Promise.resolve([roles, 1]));
-    const resp = await roleService.getAllRoles();
-    expect(resp).toEqual([roles, 1]);
+    roleService = moduleRef.get(RoleService);
+    roleRepository = moduleRef.get(RoleRepository);
+    permissionRepository = moduleRef.get(PermissionRepository);
```

Comment:
Shall we remove this since the value of permissionRepository is not used anywhere.

Reply:
Will fix service related tests in a separate PR.